### PR TITLE
Add a production targets section

### PR DESCRIPTION
**Pull request by doanac**
Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for b3ed1780dd45add20f0525c700530aa87de8fcae are browesable at: https://ci.foundries.io/projects/fio-docs/builds/814/docs/artifacts/html/index.html

File: source/reference-manual/ota/production-targets.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Before a Factory can start doing production OTAs, an initial production
+Targets file must be created. This can be done by creating a dummy wave
+like::
+
+  # fioctl wave init <wave-name> <target number> <tag>
+  fioctl wave init -k <path to offline-target-creds>.tgz populate-targets 12 production
+  fioctl wave  complete populate-targets
+
+This creates a new targets.json file for production devices subscribing
```
Comment (angolini) [inline source/reference-manual/ota/production-targets.rst:26]: `targets.json`

edit the file name using special formatting

File: source/reference-manual/ota/production-targets.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+to the ``production`` tag. It will include a single Target from CI
+build #12.
```
Comment (angolini) [inline source/reference-manual/ota/production-targets.rst:28]: a single Target from CI build (i.e.) #12.

File: source/reference-manual/ota/production-targets.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+Performing a production OTA
+---------------------------
+
+Continuing from the example above. A Factory may go through some
+development work and have a new CI build #42 that's ready to be run in
+production. A wave can be initialized with::
+
+  fioctl wave init -k <path to offline-target-creds>.tgz v2.0-update 42 production
+
+A new Wave is ready to share with device groups. A Factory might have a
```
Comment (angolini) [inline source/reference-manual/ota/production-targets.rst:40]: I'm not sure I understand why this "Wave" is capitalized

File: source/reference-manual/ota/production-targets.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+run it. However, all other production devices on the ``production`` tag
+will continue to run Target 12.
+
+Assuming the first wave to the ``canary`` device-group succeeded, an
+operator can roll out to more devices. For example a device-group named
+``us-east-1``::
+
+  fioctl waves rollout v2.0-update us-east-1
+
+Eventually the operator can update the primary targets.json file for all
```
Comment (angolini) [inline source/reference-manual/ota/production-targets.rst:62]: `targets.json`

**Review by angolini — APPROVED** [review]

File: source/reference-manual/ota/production-targets.rst (hunk 4 above)
Comment (doanac) [reply source/reference-manual/ota/production-targets.rst]: https://github.com/foundriesio/docs/pull/197/commits/d718ba310beb838e10bf806c060e3eb885e35ac5

File: source/reference-manual/ota/production-targets.rst (hunk 3 above)
Comment (doanac) [reply source/reference-manual/ota/production-targets.rst]: mistake - https://github.com/foundriesio/docs/pull/197/commits/d718ba310beb838e10bf806c060e3eb885e35ac5

File: source/reference-manual/ota/production-targets.rst (hunk 2 above)
Comment (doanac) [reply source/reference-manual/ota/production-targets.rst]: i re-worded - https://github.com/foundriesio/docs/pull/197/commits/d718ba310beb838e10bf806c060e3eb885e35ac5

File: source/reference-manual/ota/production-targets.rst (hunk 1 above)
Comment (doanac) [reply source/reference-manual/ota/production-targets.rst]: https://github.com/foundriesio/docs/pull/197/commits/d718ba310beb838e10bf806c060e3eb885e35ac5

**Comment by doanac** [comment]
Docs for d718ba310beb838e10bf806c060e3eb885e35ac5 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/821/docs/artifacts/html/index.html

File: source/reference-manual/ota/production-targets.rst [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Production targets are managed by using Waves. Normally, when the
+TUF targets metadata is updated, all devices see it and can start
+updating themselves. Waves are a feature that allow Factory operators
+to control the rate at which devices see a new copy of the TUF targets
+metadata.
+
+Before a Factory can start doing production OTAs, an initial production
+Targets file must be created. This can be done by creating a dummy wave
+like::
```
Comment (MatthewCroughan) [inline source/reference-manual/ota/production-targets.rst:20]:
```suggestion
::
```

**Review by MatthewCroughan — APPROVED** [review]
Love this, really simple.

**Comment by doanac** [comment]
Docs for 4fd4d564e47c0b61d09fddf737026ca1804a6b8a are browesable at: https://ci.foundries.io/projects/fio-docs/builds/832/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 4fd4d564e47c0b61d09fddf737026ca1804a6b8a are browesable at: https://ci.foundries.io/projects/fio-docs/builds/833/docs/artifacts/html/index.html

**Review by angolini — APPROVED** [review]
@MatthewCroughan, do you think "waves" should be added to the terminology page?

**Comment by MatthewCroughan** [comment]
@angolini Yeah, in a separate PR. https://github.com/foundriesio/docs/issues/201

**Comment by doanac** [comment]
So this is good to squash and merge then?

**Comment by MatthewCroughan** [comment]
Yup, LGTM after squash. Funny enough, I would have the option to do that but it's disabled for this repo, probably for good reason. @rsalveti any ideas?
image

**Comment by doanac** [comment]
i like squashing locally. it allows my to produce gpg signed commits.

**Comment by doanac** [comment]
rebased on master and squashed

**Comment by doanac** [comment]
Docs for 8d0270147f8ed380dfd36994c58400b32fb82630 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/842/docs/artifacts/html/index.html

**Pull request merged**